test_libs/pyspec/eth2spec/utils/ssz/ssz_impl.py

```diff
@@ -14,68 +10,47 @@
 BYTES_PER_LENGTH_OFFSET = 4
 
 
-def is_basic_type(typ):
-    return is_uint_type(typ) or is_bool_type(typ)
-
-
-def serialize_basic(value, typ):
-    if is_uint_type(typ):
-        return value.to_bytes(uint_byte_size(typ), 'little')
-    elif is_bool_type(typ):
+def serialize_basic(value: SSZValue):
+    if isinstance(value, uint):
+        return value.to_bytes(value.type().byte_len, 'little')
+    elif isinstance(value, Bit):
         if value:
             return b'\x01'
         else:
             return b'\x00'
     else:
-        raise Exception("Type not supported: {}".format(typ))
+        raise Exception(f"Type not supported: {type(value)}")
 
 
-def deserialize_basic(value, typ):
-    if is_uint_type(typ):
+def deserialize_basic(value, typ: BasicType):
+    if issubclass(typ, uint):
         return typ(int.from_bytes(value, 'little'))
-    elif is_bool_type(typ):
+    elif issubclass(typ, Bit):
         assert value in (b'\x00', b'\x01')
-        return True if value == b'\x01' else False
+        return Bit(value == b'\x01')
```

Comment:
Can use `typ` instead of `Bit` here, right?
Is there a reason to not be consistent with `uint` by using `typ`?

Reply:
good spot. `typ` is generally better, as it creates the proper subclass (e.g. byte, instead of uint8). But with coercion, it's eventually put into the right type. Will fix this though.

Reply:
see c73417b